sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -195,12 +196,33 @@ class AstBuilder extends DataTypeAstBuilder
       case _ =>
     }
 
-    val labelText = beginLabelCtx.
+    beginLabelCtx.
       map(_.multipartIdentifier().getText).getOrElse(java.util.UUID.randomUUID.toString).
       toLowerCase(Locale.ROOT)
+  }
+
+
+
+  override def visitBeginEndCompoundBlock(ctx: BeginEndCompoundBlockContext): CompoundBody = {
+    val labelText = generateLabelText(Option(ctx.beginLabel()), Option(ctx.endLabel()))
+
     visitCompoundBodyImpl(ctx.compoundBody(), Some(labelText), allowVarDeclare = true)
   }
 
+  override def visitWhileStatement(ctx: WhileStatementContext): WhileStatement = {
```

Comment:
nit: let's keep the same order of statements as in `SqlBaseParser.g4`, i.e. move while statement after if else